test/e2e/aws_test.go

```diff
@@ -45,101 +47,285 @@ import (
-
-			By("Deploying CNI to created Cluster")
-			deployCNI(testTmpDir, namespace, clusterName, *cniManifests)
-			waitForMachineNodeReady(namespace, machineName)
+			By("Creating a cluster with single control plane")
+			makeSingleControlPlaneCluster(setup)
 
 			By("Creating the second Control Plane Machine")
-			awsMachineName = cpAWSMachinePrefix + "-1"
-			bootstrapConfigName = cpBootstrapConfigPrefix + "-1"
-			machineName = cpMachinePrefix + "-1"
-			createAdditionalControlPlaneMachine(namespace, clusterName, machineName, awsMachineName, bootstrapConfigName, instanceType)
+			awsMachineName := setup.cpAWSMachinePrefix + "-1"
+			bootstrapConfigName := setup.cpBootstrapConfigPrefix + "-1"
+			machineName := setup.cpMachinePrefix + "-1"
+			createAdditionalControlPlaneMachine(setup, machineName, awsMachineName, bootstrapConfigName)
 
 			By("Creating the third Control Plane Machine")
-			awsMachineName = cpAWSMachinePrefix + "-2"
-			bootstrapConfigName = cpBootstrapConfigPrefix + "-2"
-			machineName = cpMachinePrefix + "-2"
-			createAdditionalControlPlaneMachine(namespace, clusterName, machineName, awsMachineName, bootstrapConfigName, instanceType)
+			awsMachineName = setup.cpAWSMachinePrefix + "-2"
+			bootstrapConfigName = setup.cpBootstrapConfigPrefix + "-2"
+			machineName = setup.cpMachinePrefix + "-2"
+			createAdditionalControlPlaneMachine(setup, machineName, awsMachineName, bootstrapConfigName)
 
 			By("Creating the MachineDeployment")
-			createMachineDeployment(namespace, clusterName, machineDeploymentName, awsMachineTemplateName, mdBootstrapConfig, instanceType, initialReplicas)
+			createMachineDeployment(setup)
 
 			By("Scale the MachineDeployment up")
-			scaleMachineDeployment(namespace, machineDeploymentName, initialReplicas, scaleUpReplicas)
+			scaleMachineDeployment(setup.namespace, setup.machineDeploymentName, setup.initialReplicas, scaleUpReplicas)
 
 			By("Scale the MachineDeployment down")
-			scaleMachineDeployment(namespace, machineDeploymentName, scaleUpReplicas, scaleDownReplicas)
+			scaleMachineDeployment(setup.namespace, setup.machineDeploymentName, scaleUpReplicas, scaleDownReplicas)
 
 			By("Scale the MachineDeployment up again")
-			scaleMachineDeployment(namespace, machineDeploymentName, scaleDownReplicas, initialReplicas)
+			scaleMachineDeployment(setup.namespace, setup.machineDeploymentName, scaleDownReplicas, setup.initialReplicas)
 
 			By("Deleting the Cluster")
-			deleteCluster(namespace, clusterName)
+			deleteCluster(setup.namespace, setup.clusterName)
 
 		})
 	})
+
+	Describe("multiple workload clusters", func() {
+		Context("in different namespaces", func() {
+			var ns1, clName1, ns2, clName2 string
+			It("should create first cluster", func() {
+				ns1 = setup.namespace
+				clName1 = setup.clusterName
+				By("Creating first cluster with single control plane")
+				makeSingleControlPlaneCluster(setup)
```

Comment:
We seem to be doing this step in all of the different jobs now. I'm wondering if this should be moved to a BeforeEach/AfterEach instead, if we have any jobs that would require a different setup, they could be moved to a separate test file.

Not blocking for this PR, though.

Reply:
Sounds good. Will do it in a different PR